src/strategy.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {MinimalRenamer} from './minimal-renamer';
+import {type SkipPredicate} from './skip';
+
+/**
+ * Renaming function.
+ *
+ * A renaming function must satisfy the following:
+ * - A renaming function always produces the same output for a specific input.
+ * - A renaming function does not produce anything that is in the user-specified
```

Comment:
After pondering about this for a while. I think this bulletpoint isn't really necessary. The "minimal renamer" is the only strategy that uses the skiplist, and even then it doesn't use a custom renaming function when given one.

So I think it is better to omit this rule, and update the documentation to mention that the only requirement is for the function to be idempotent